pool/app/pool_common/entity_base_model.ml

```diff
@@ -0,0 +1,132 @@
+module Boolean = struct
```

Comment:
Could you make these module implementations have module type `BooleanSig`? Via `module Boolean : BooleanSig = struct...`

And please do it for all other base modules

Reply:
Resolving. I didn't manage to create a module using the with keyword.